src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,182 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var extendFlat = require('../../lib/extend').extendFlat;
+var colorAttributes = require('../../components/colorscale/attributes');
+
+var scatterAttrs = require('../scatter/attributes');
+var scatterMarkerAttrs = scatterAttrs.marker;
+var colorbarAttrs = require('../../components/colorbar/attributes');
+
+var marker = extendFlat({
+    editType: 'calc'
+}, colorAttributes('marker', {editType: 'calc'}),
+    {
+        showscale: scatterMarkerAttrs.showscale,
+        colorbar: colorbarAttrs,
+        shape: {
+            valType: 'enumerated',
+            values: ['straight', 'curved'],
+            dflt: 'curved',
```

Comment:
Perhaps to match scatter we should have `'linear'` and `'spline'`? Though this curve is really a different thing, as it's constrained to be horizontal at both ends - an option we've discussed adding to scatter lines in fact - (related https://github.com/plotly/plotly.js/issues/993 talks about monotone splines, which is yet another variant) ... but if we can decide on a name this should have in scatter, it would be good to use the same name here. What about `'hspline'` for "spline with horizontal ends" (and we'd also add `'vspline'` to scatter, though it wouldn't be added here unless we make a vertical `orientation` option for this trace type.)

Also, I wonder if the default should be `'linear'`? Curved matches `sankey`, but straight matches `parcoords` and I think it actually makes connections a bit easier to follow - which may be why the existing implementations you showed use straight connections.

The other visual changes you made vs the existing implementations, which seem to come from `sankey` - a bar of color instead of a black line, a halo around the category labels - are clear improvements!

Reply:
'linear' + 'hspline' with 'linear' as default sounds good to me!

Reply:
Done in https://github.com/jonmmease/plotly.js/pull/1/commits/7b751009fc9804272316f0bb539ed0386c0858bd